pkg/cache/listers.go

```diff
@@ -27,21 +27,29 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
-// NewGenericClusterLister creates a new instance for the GenericClusterLister.
-func NewGenericClusterLister(indexer cache.Indexer, resource schema.GroupResource) *GenericClusterLister {
-	return &GenericClusterLister{
+// NewGenericClusterLister creates a new instance for the ClusterLister.
+func NewGenericClusterLister(indexer cache.Indexer, resource schema.GroupResource) *ClusterLister {
+	return &ClusterLister{
 		indexer:  indexer,
 		resource: resource,
 	}
 }
 
-// GenericClusterLister is a lister that supports multiple logical clusters. It can list the entire contents of the backing store, and return individual cache.GenericListers that are scoped to individual logical clusters.
-type GenericClusterLister struct {
+// GenericClusterLister is a lister skin on a generic Indexer
+type GenericClusterLister interface {
```

Comment:
Does any of our framework/library code need this? Or if it does, can we declare it where it's used?

Reply:
This (colocating the interface with the impl) is just parallel to how k8s does it, and I think there's more utility to being parallel than doing the "interface at the callsite" thing. Although, in k8s the library is `k8s.io/client-go/tools/cache`. I don't have strong feelings on where this lives but please keep in mind that every day spent on this sort of stuff incurs large rebase cost for the kcp refactor PR.